### ISSUE TITLE
Menu.Item with Polymorphic component: when using keyboard arrows to enter Menu.Dropdown, I get the error TypeError: Cannot read properties of undefined (reading 'focus')

### ISSUE THREAD

**Issue by QuentinLucas** [mantinedev/mantine#4748]
### What package has an issue

@mantine/core

### Describe the bug

**Context:**

To implement a navigation bar, we use Mantine's `Menu` component.
Some sections have sub-sections items. To implement them, we use the Mantine's `Menu` dropdown.
We show a dropdown to display the sub-sections when clicking on a section.

**The issue:**

The navigation with the mouse works without issue.
However, when you navigate with the keyboard, press RETURN key on the focused section button, and use the DOWN ARROW key to enter (and focus) the dropdown, an error appears.

<img width="964" alt="Capture d’écran 2023-08-25 à 14 10 03" src="https://github.com/mantinedev/mantine/assets/22346872/ae9f9ca4-ee25-490f-b93e-1cef5fbd487b">

The `Menu` code looks like following (the code has been simplified for the issue)

```typescript
<Menu>
    <Menu.Target>
        <MantineButton />
    </Menu.Target>
    <Menu.Dropdown>
        <Menu.Item
              component={CustomLink}
              href={href}
        >
           Link text
        </Menu.Item>
    </Menu.Dropdown>
</Menu>
``` 

```typescript
const CustomLink = ({
    children,
    href,
}: PropsWithChildren<CustomLinkProps>): JSX.Element => {
    return (
        <Text
	    component={"a"}
       	    href={href}
        >
            {children}
        </Text>
    );
};
```

Using debugger, you get `wrapperRef.current.querySelectorAll("[data-menu-item]")[0]` as undefined.

<img width="1123" alt="Capture d’écran 2023-09-11 à 21 42 13" src="https://github.com/mantinedev/mantine/assets/22346872/837a2077-28da-4e13-bbfd-1776d7aa5460">

**What we conclude:**
➡️ The lib cannot find the component with the selector **data-menu-item** when pressing arrow key.
The component that plays the role of `Menu.Item` needs the **data-menu-item** attribute to allow the keyboard navigation work and not raise an error.

Indeed, if we add **data-menu-item** attribute to the `CustomLink`, the error disappears:

```typescript
const CustomLink = ({
    children,
    href,
}: PropsWithChildren<CustomLinkProps>): JSX.Element => {
    return (
        <Text
	    component={"a"}
       	    href={href}
            data-menu-item
        >
            {children}
        </Text>
    );
};
```

**What we did as fix:**

```typescript
import NextLink from 'next/link';

const CustomLink = ({
    children,
    href,
    ...rest
}: PropsWithChildren<CustomLinkProps>): JSX.Element => {
    return (
        <Text
	    component={"a"}
       	    href={href}
            {...rest}
        >
            {children}
        </Text>
    );
};
```

In our case, we have to pass all the residual props in the `CustomLink` components.

In reality, it was more complex to debug because we have an intermediate component for tracking (TrackedCustomLink) which wraps CustomLink.

**Time lost:**
We wasted 2x1h to find a solution to our problem.

**The problem I am raising is the following:**
- If we are using Polymorphic components with custom component, AND we forget to pass the residual props in the context of a `Menu.Item`, we get the error
- Easy to forget the residual props, no type error
- The Menu.item documentation [Menu](https://mantine.dev/core/menu/) does not mention the **data-menu-item**
- The Polymorphic components documentation [Polymorphic](https://mantine.dev/guides/polymorphic/) does not mention the subtlety of residual props

Moreover, for the `v6.0.17`, within the commit https://github.com/mantinedev/mantine/commit/f138de508fa89bcd4b85e4fd3ba6602a51211797, the following change has been added:
<img width="798" alt="Capture d’écran 2023-09-11 à 22 03 22" src="https://github.com/mantinedev/mantine/assets/22346872/4104326b-05f3-4d4e-b9cf-0abdd56e5583">
That means the error "TypeError: Cannot read properties of undefined (reading 'focus')" may no longer appear, yielding a more complex debugging when raising the error.

### What version of @mantine/hooks page do you have in package.json?

"@mantine/core": "6.0.8"

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/zealous-bohr-9kp63f?file=/src/App.tsx

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

**In documentation**
- I suggest adding a section to the documentation [Polymorphic, with-other-components](https://mantine.dev/guides/polymorphic/#with-other-components) when using Polymorphic component with custom components, to emphasise that certain props are necessary for Mantine components to work properly.
- Maybe also add a hint for the data-menu-item props in the documentation [Menu](https://mantine.dev/core/menu/)

Perhaps this issue will lead to better ideas, with a code approach.

Thank you for reading the issue.

**Comment by rtivital** [comment]
Documentation was updated, it will be available with 7.0.1 patch

<img width="656" alt="Screenshot" src="https://github.com/mantinedev/mantine/assets/10353856/ae3bf088-7c97-4a1b-b2fc-897b7e36dcb5">

**Comment by rtivital** [comment]
The issue is resolved in [7.0.1](https://github.com/mantinedev/mantine/releases/tag/7.0.1)
